### PR TITLE
Tina studio2019

### PR DESCRIPTION
**Pull request by TinaCaputa**
Improve the process of using project templates 
Use IMessageBox interface to implement the MessageBoxService and display messages.
Move all UI messages to PluginResources.resx file
Update the progress bar: show it only when the project is creating
Fix issue: notifications are duplicated when pressing on "Check Project Request"
Align the delete icon to center
Refactor code: split the methods on which i've added fixes into small methods

### CODE REVIEW

File: InSource/Helpers/Constants.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Sdl.Community.InSource.Helpers
+{
+	public class Constants
+	{
+		public readonly string NotificationGroupId = "b0261aa3-b6a5-4f69-8f94-3713784ce8ef";
+		public readonly string CreateProjectText = "Create project";
```
Comment (andreaGhisa) [inline InSource/Helpers/Constants.cs:6]: Move this into resource file

File: InSource/Helpers/Constants.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Sdl.Community.InSource.Helpers
+{
+	public class Constants
+	{
+		public readonly string NotificationGroupId = "b0261aa3-b6a5-4f69-8f94-3713784ce8ef";
```
Comment (andreaGhisa) [inline InSource/Helpers/Constants.cs:5]: move this into notification related logic.

File: InSource/Helpers/Constants.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Sdl.Community.InSource.Helpers
+{
+	public class Constants
```
Comment (andreaGhisa) [inline InSource/Helpers/Constants.cs:3]: we don't need this class

File: InSource/ProjectCreator.cs [hunk 4]
```diff
@@ -180,6 +204,26 @@ private string GetProjectFolderPath(string name,string pathToTemplate)
...
         }
 
+		// Validate if any language direction is configured on the custom template
+		private bool IsAnyLanguageDirection(string templatePath)
+		{
+			try
+			{
+				var templateXml = XElement.Load(templatePath);
+				var languageDirection = templateXml.Descendants("LanguageDirections").Descendants("LanguageDirection")?.FirstOrDefault();				
+				if (languageDirection == null)
```
Comment (andreaGhisa) [inline InSource/ProjectCreator.cs:214]: This condition and return can be refactored.

Comment (TinaCaputa) [reply InSource/ProjectCreator.cs]: Refactored the code: 
image

File: InSource/Helpers/Constants.cs (hunk 3 above)
Comment (TinaCaputa) [reply InSource/Helpers/Constants.cs]: I've removed the class and moved fields

File: InSource/Helpers/Constants.cs (hunk 2 above)
Comment (TinaCaputa) [reply InSource/Helpers/Constants.cs]: Done. the NotificationGroupId was moved from the Constants class

File: InSource/Helpers/Constants.cs (hunk 1 above)
Comment (TinaCaputa) [reply InSource/Helpers/Constants.cs]: Done. the CreateProjectText and CreateNewProject text were moved to PluginResources.resx file

File: InSource/InSourceViewControl.cs [hunk 5]
```diff
@@ -34,6 +35,16 @@ public InSourceViewControl()
...
 		}
 
+		public void ClearMessages()
+		{
+			_resultsTextBox.Text = "";
+		}
+
+		public void ReportMessage(FileBasedProject fileBasedProject, string message)
+		{
+			_resultsTextBox.AppendText("\r\n" + message);
```
Comment (andreaGhisa) [inline InSource/InSourceViewControl.cs:45]: Can we use here string interpolation?

File: InSource/InSourceViewControl.cs [hunk 6]
```diff
@@ -34,6 +35,16 @@ public InSourceViewControl()
 			_messageBoxService = new MessageBoxService();
```
Comment (andreaGhisa) [inline InSource/InSourceViewControl.cs:35]: this can be made read only.

Comment (TinaCaputa) [reply InSource/InSourceViewControl.cs:34]: Done. I've update within the new commit

File: InSource/InSourceViewControl.cs (hunk 5 above)
Comment (TinaCaputa) [reply InSource/InSourceViewControl.cs:44]: Done. I've updated within the new commit.

File: InSource/InSourceViewControl.cs [hunk 7]
```diff
@@ -34,6 +35,16 @@ public InSourceViewControl()
...
 		}
 
+		public void ClearMessages()
```
[the remark this answers is not in the document]
Comment (TinaCaputa) [reply InSource/InSourceViewControl.cs:37]: Done, I've update the code within the new commit

File: InSource/TimerControl.cs [hunk 8]
```diff
@@ -3,150 +3,179 @@
...
+				refreshIntervalLbl.ForeColor = Color.Gray;
+				_timerSettings.HasTimer = false;
+				_timerSettings.Minutes = 0;
+
+				Timer.Enabled = false;
+				SetRemainingLabel(PluginResources.DisabledTimer_Message, Color.Gray);
+			}
+		}
+
+		private void SetIntervalTextBox(bool isEnabled, string text)
```
Comment (andreaGhisa) [inline InSource/TimerControl.cs:157]: This method can be refactored to receive a Label and it can be used for refreshIntervalLbl

Comment (TinaCaputa) [reply InSource/TimerControl.cs:158]: The changes can be checked in the new commit

**Pull request merged**